src/NSubstitute/Routing/Handlers/EventSubscriptionHandler.cs

```diff
@@ -46,7 +46,7 @@ private static void If(ICall call, Func<ICall, Predicate<EventInfo>> meetsThisSp
             var matchingEvent = GetEvents(call, meetsThisSpecification).FirstOrDefault();
             if (matchingEvent != null)
             {
-                takeThisAction(matchingEvent.Name, call.GetArguments()[0]);
+                takeThisAction(matchingEvent.Name, call.GetOriginalArguments()[0]);
```

Comment:
Maybe add this commit comment to the code too? (see it doesn't get changed back in a later commit?)

Reply:
Good point! Did it for all the non-obvious places around the PR.